ntoskrnl/cc/copy.c

```diff
@@ -809,48 +783,51 @@ CcZeroData (
         IO_STATUS_BLOCK Iosb;
         KEVENT Event;
         PMDL Mdl;
-        ULONG i;
         ULONG CurrentLength;
-        PPFN_NUMBER PfnArray;
+        PHYSICAL_ADDRESS ZeroAddress;
+        PHYSICAL_ADDRESS MaxAddress;
 
-        /* Setup our Mdl */
-        Mdl = IoAllocateMdl(NULL, min(Length, MAX_ZERO_LENGTH), FALSE, FALSE, NULL);
+        /* Get a zeroed MDL */
+        ZeroAddress.QuadPart = 0;
+        MaxAddress.QuadPart = MAXLONGLONG;
+        Mdl = MmAllocatePagesForMdl(ZeroAddress, MaxAddress, ZeroAddress, min(Length, MAX_ZERO_LENGTH));
```

Comment:
Hm, MSDN recommends using `MmAllocatePagesForMdlEx`. Does it make sense here?

Reply:
That would be with 0 as flags. That's still better I guess.